OpenRA.Mods.Common/Widgets/Logic/Ingame/CommandBarLogic.cs

```diff
@@ -58,14 +55,8 @@ public CommandBarLogic(Widget widget, World world)
 				attackMoveButton.IsHighlighted = () => world.OrderGenerator is GenericSelectTarget
 					&& ((GenericSelectTarget)world.OrderGenerator).OrderName == "AttackMove";
 
-				attackMoveButton.OnClick = () =>
-				{
-					if (attackMoveButton.IsHighlighted())
```

Comment:
It should be possible to have the best of both worlds by doing something like:
```csh
Action<bool> toggle = isRepeat => 
{
	if (attackMoveButton.IsHighlighted() && !isRepeat)
		world.CancelInputMode();
	else
		world.OrderGenerator = new GenericSelectTarget(selectedActors,
			"AttackMove", "attackmove", Game.Settings.Game.MouseButtonPreference.Action);
};

attackMoveButton.OnClick = () => toggle(false);
attackMoveButton.OnKeyPress = e => toggle(e.IsRepeat);
```

You will also need to do the same change to the `guardButton` for consistency.